ec2-worker/iam.tf

```diff
@@ -41,10 +41,10 @@ data "aws_iam_policy_document" "concourse_worker_policy" {
 }
 
 resource "aws_iam_role_policy" "concourse_worker_policy" {
-  count  = "${length(var.cross_account_worker_role_arn) > 0 ? 0 : 1}"     # Disable if accessing another AWS account through an assume role
+  count  = coalesce(var.cross_account_worker_role_arn, 1) # Disable if accessing another AWS account through an assume role
```

Comment:
I don't think you'll get the expected result with this `coalesce`. If `var.cross_account_worker_role_arn` is not null, it'll be a string, but `count` can't be a string so terraform will probably return an error.

You want to set `count` to 0 if `var.cross_account_worker_role_arn` is not null.